src/Workspaces/VisualBasic/Portable/CodeCleanup/Providers/ReduceTokensCodeCleanupProvider.vb

```diff
@@ -222,26 +264,75 @@ Namespace Microsoft.CodeAnalysis.CodeCleanup.Providers
                 Return valueText
             End Function
 
-            Private Shared Function CreateFloatingLiteralToken(token As SyntaxToken, newValueString As String, newValue As Double) As SyntaxToken
-                Debug.Assert(token.Kind = SyntaxKind.FloatingLiteralToken)
+            Private Function GetIntegerLiteralValueString(value As Object, base As LiteralBase) As String
+                Select Case base
+                    Case LiteralBase.Decimal
+                        Return CType(value, ULong).ToString(CultureInfo.InvariantCulture)
+                    Case LiteralBase.Hexadecimal
+                        Dim valueString = ""
+                        If TypeOf (value) Is Short Then
```

Comment:
I've unified this into a single helper. Still need to convert to Short -> UShort and Integer -> UInteger first otherwise the conversion appears to be setting the leading bits for negative numbers converted directly to ULong. I'll put a comment in though to make it obvious why it needs to be done that way